### PR TITLE
Default requests to limits in Workload

### LINKED ISSUE #316: Default requests to limits in Workload
**What would you like to be added**:

If limit is specified, but the corresponding request is not, set request equal to limit.

**Why is this needed**:

This is the defaulting that k8s API does for Pods, but that wouldn't apply until the Pods are created.

https://github.com/kubernetes/kubernetes/blob/42a5eb48184547465ef4be962af2a07422311f82/pkg/apis/core/v1/defaults.go#L149-L177

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Default requests to limits if requests are not set.
This matches the internal defaulting for k8s Pods.

#### Which issue(s) this PR fixes:

Fixes #316

### CODE REVIEW

File: apis/kueue/webhooks/workload_webhook.go [hunk 1]
```diff
@@ -57,10 +58,28 @@ func (w *WorkloadWebhook) Default(ctx context.Context, obj runtime.Object) error
...
+		setContainersDefaults(podSet.Spec.InitContainers)
+		setContainersDefaults(podSet.Spec.Containers)
 	}
 	return nil
 }
 
+func setContainersDefaults(containers []corev1.Container) {
+	for i := range containers {
+		c := &containers[i]
+		if c.Resources.Limits != nil {
```
Comment (kerthcet) [inline apis/kueue/webhooks/workload_webhook.go:70]: Not related to this PR, we currently leverage the requests in resource quota, should we also consider the limits.

Comment (alculquicondor) [reply apis/kueue/webhooks/workload_webhook.go:70]: Are you talking about kubernetes' ResourceQuotas? If so, I would imagine they run after defaults are applied, but I don't have expertise to know for sure.

Comment (kerthcet) [reply apis/kueue/webhooks/workload_webhook.go:70]: I mean the resources.Limits in podSpec.

Comment (alculquicondor) [reply apis/kueue/webhooks/workload_webhook.go:70]: It was the intention that we only look at requests, similar to the NodeResources Filter plugin in kube-scheduler.

**Pull request merged**